third_party/txt/src/txt/paragraph.cc

```diff
@@ -556,58 +547,78 @@ void Paragraph::Layout(double width, bool force) {
             builder.allocRunPos(paint, glyph_blob.end - glyph_blob.start);
 
         for (size_t glyph_index = glyph_blob.start;
-             glyph_index < glyph_blob.end; ++glyph_index) {
-          size_t blob_index = glyph_index - glyph_blob.start;
-          blob_buffer.glyphs[blob_index] = layout.getGlyphId(glyph_index);
-
-          size_t pos_index = blob_index * 2;
-          double glyph_x_offset = layout.getX(glyph_index) + justify_x_offset;
-          blob_buffer.pos[pos_index] = glyph_x_offset;
-          blob_buffer.pos[pos_index + 1] = layout.getY(glyph_index);
-
-          // The glyph may be a ligature.  Determine how many input characters
-          // are joined into this glyph.  Note that each character may be
-          // encoded as multiple UTF-16 code units.
-          Range<int32_t> glyph_code_units;
-          std::vector<size_t> subglyph_code_unit_counts;
+             glyph_index < glyph_blob.end;) {
+          size_t cluster_start_glyph_index = glyph_index;
+          uint32_t cluster = layout.getGlyphCluster(cluster_start_glyph_index);
+          double glyph_x_offset;
```

Comment:
Looks like we're guaranteed to initialise on line 566, but I wonder if it's worth defensively initialising for future-proofing? I suppose any badness should be relatively obvious regardless.